crates/services/compression/src/ports/block_source.rs

```diff
@@ -1,11 +1,41 @@
-/// Type alias for fuel_core_types::blockchain::block::Block
-pub type Block = fuel_core_types::blockchain::block::Block;
+use fuel_core_services::stream::BoxStream;
+
+/// Block with metadata needed for compression
+#[cfg_attr(any(test, feature = "test-helpers"), derive(Default))]
+#[derive(Clone, Debug)]
+pub struct BlockWithMetadata {
+    block: fuel_core_types::blockchain::block::Block,
+    // perhaps slice with lifetime?
+    events: Vec<fuel_core_types::services::executor::Event>,
```

Comment:
Wouldn't mind a slice with lifetimes here, but feels like it could be unergonomic - so perhaps let's start without it and optimize later.

```suggestion
    events: Vec<fuel_core_types::services::executor::Event>,
```

Reply:
I don't think we need an issue to track this. It doesn't seem like a super important optimization.

Reply:
https://github.com/FuelLabs/fuel-core/issues/2870 and in code: https://github.com/FuelLabs/fuel-core/pull/2844/commits/865df644f2c15c3a6d6f728d633199c77075c856#diff-b35ece6e136e2fe3793da56f4456d50692646e9d371946e26024ada643151182R8